tools/c7n_azure/c7n_azure/lookup.py

```diff
@@ -0,0 +1,68 @@
+# 2019 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import jmespath
+from jsonschema import validate, exceptions
+
+
+class Lookup(object):
+    RESOURCE_SOURCE = 'resource'
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'source': {'type': 'string', 'enum': ['resource']},
+            'key': {'type': 'string'},
+            'default-value': {'oneOf': [
+                {'type': 'string'},
+                {'type': 'number'},
+                {'type': 'boolean'}
+            ]}
+        },
+        'required': ['source', 'key']
+    }
+
+    @staticmethod
+    def extract(source, data=None):
+        if not Lookup.is_lookup(source):
+            return source
+        else:
+            return Lookup.get_value(source, data)
+
+    @staticmethod
+    def is_lookup(source):
+        if type(source) is not dict:
+            return False
+
+        try:
+            validate(instance=source, schema=Lookup.schema)
```

Comment:
Why do we need validate there? IIRC, c7n builds giant schema to validate entire policy, so if policy passed the schema, it should pass this validation too.

Reply:
Discussed why it was here offline, but decided to remove since it was a developer check and not user check :)